### PR TITLE
fix(ci): layer rename

### PR DESCRIPTION
**Pull request by sthulb**
**Issue number:** #5227

### Changes

> Please provide a summary of what's being changed

Provides a workflow to rename layers to include the correct architecture

### User experience

> Please share what the user experience looks like before and after this change

## Checklist

If your change doesn't seem to apply, please leave them unchecked.

* [X] [Meet tenets criteria](https://docs.powertools.aws.dev/lambda/python/#tenets)
* [X] I have performed a self-review of this change
* [X] Changes have been tested
* [X] Changes are documented
* [X] PR title follows [conventional commit semantics](https://github.com/aws-powertools/powertools-lambda-python/blob/develop/.github/semantic.yml)

<details>
<summary>Is this a breaking change?</summary>

**RFC issue number**:

Checklist:

* [ ] Migration process documented
* [ ] Implement warnings (if it can live side by side)

</details>

## Acknowledgment

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

**Disclaimer**: We value your time and bandwidth. As such, any pull requests created on non-triaged issues might not be successful.

### CODE REVIEW

File: .github/workflows/layer_rename.yml [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+  workflow_dispatch:
+    inputs:
+      environment:
+        description: Deployment environment
+        type: choice
+        options:
+          - beta
+          - prod
+        default: Gamma
+        required: true
```
Comment (leandrodamascena) [inline .github/workflows/layer_rename.yml:20]: Do we need this?

File: .github/workflows/layer_rename.yml [hunk 2]
```diff
@@ -0,0 +1,161 @@
...
+        description: Layer version to duplicate
+        type: number
+        required: true
+  workflow_call:
+    inputs:
+      environment:
+        description: Deployment environment
+        type: string
+        default: Gamma
+        required: true
```
Comment (leandrodamascena) [inline .github/workflows/layer_rename.yml:31]: Do we need this?

**Review by leandrodamascena — CHANGES_REQUESTED** [review]

File: .github/workflows/layer_rename.yml (hunk 2 above)
Comment (sthulb) [reply .github/workflows/layer_rename.yml:31]: Yes – we still use dynamic environment loading

File: .github/workflows/layer_rename.yml (hunk 1 above)
Comment (sthulb) [reply .github/workflows/layer_rename.yml:20]: Yes – we still use dynamic environment loading

Comment (leandrodamascena) [reply .github/workflows/layer_rename.yml:20]: I see it on line 126. But we are renaming prod to prod, we don't rename prod to beta for example, we don't need it. But anyway, please let me know if you want to keep this to run the pipeline and I'll approve.

**Review by leandrodamascena — APPROVED** [review]

**Pull request merged**